sanic_routing/tree.py

```diff
@@ -57,186 +88,230 @@ def display(self) -> None:
             child.display()
 
     def render(self) -> t.Tuple[t.List[Line], t.List[Line]]:
-        output: t.List[Line] = []
+        # output - code injected into the source as it is being
+        #    called/evalueated
+        # delayed - code that is injected after you do all of its children
+        #    first
+        # final - code that is injected at the very end of all rendering
+        src: t.List[Line] = []
         delayed: t.List[Line] = []
         final: t.List[Line] = []
 
         if not self.root:
-            output, delayed, final = self.to_src()
+            src, delayed, final = self.to_src()
         for child in self.children.values():
             o, f = child.render()
-            output += o
+            src += o
             final += f
-        return output + delayed, final
-
-    def apply_offset(self, amt, apply_self=True, apply_children=False):
-        if apply_self:
-            self.offset += amt
-        if apply_children:
-            for child in self.children.values():
-                child.apply_offset(amt, apply_children=True)
+        return src + delayed, final
 
     def to_src(self) -> t.Tuple[t.List[Line], t.List[Line], t.List[Line]]:
-        indent = (self.level + 1) * 2 - 3 + self.offset
+        siblings = self.parent.children if self.parent else {}
+        first_sibling: t.Optional[Node] = None
+
+        if not self.first:
+            first_sibling = list(siblings.values())[0]
```

Comment:
I suppose this is not performance critical but maybe `next(iter(siblings.values()))` to avoid making a full list.

Reply:
That makes sense.